cli/file_watcher.rs

```diff
@@ -28,43 +83,51 @@ async fn error_handler(watch_future: WatchFuture) {
 }
 
 pub async fn watch_func<F>(
-  watch_paths: &[PathBuf],
+  paths: &[PathBuf],
   closure: F,
 ) -> Result<(), AnyError>
 where
   F: Fn() -> WatchFuture,
 {
+  let (_watcher, receiver) = new_watcher(paths)?;
+  let debounce = Mutex::new(Debounce::new(receiver, DEBOUNCE_TIMEOUT_MS));
   loop {
     let func = error_handler(closure());
-    let mut is_file_changed = false;
-    select! {
-      _ = file_watcher(watch_paths) => {
-          is_file_changed = true;
-          info!(
-            "{} File change detected! Restarting!",
-            colors::intense_blue("Watcher")
-          );
-        },
-      _ = func => { },
-    };
-    if !is_file_changed {
-      info!(
-        "{} Process terminated! Restarting on file change...",
-        colors::intense_blue("Watcher")
-      );
-      file_watcher(watch_paths).await?;
-      info!(
-        "{} File change detected! Restarting!",
-        colors::intense_blue("Watcher")
-      );
+    func.await;
+    info!(
+      "{} Process terminated! Restarting on file change...",
+      colors::intense_blue("Watcher")
+    );
+    wait_for_file_change(&debounce).await?;
```

Comment:
@bartOssh this doesn't look right - if `func` never finishes then process won't be restarted on file change - eg. if you have a server then `func.await` will never finish. `select!` is required here

Reply:
This was something I was warring about, but wasn't sure how to fix. I will try to rethink it and bring a solid solution. thx

Reply:
@bartOssh I think previous version using `select!` was fine :)